tests/unit/views/booking/Book.spec.js

```diff
@@ -0,0 +1,277 @@
+import { shallowMount } from '@vue/test-utils';
+import { expect } from 'chai';
+
+import BookingNavigation from '@/components/booking/BookingNavigation.vue';
+import ProductionBanner from '@/components/production/ProductionBanner.vue';
+import { swal } from '@/utils';
+import Book from '@/views/booking/Book.vue';
+import stages from '@/views/booking/bookingStages';
+
+import FakePerformance from '../../fixtures/FakePerformance';
+import {
+  executeWithServer,
+  generateMountOptions,
+  runApolloQuery,
+  seedAndAuthAsUser,
+  waitFor,
+} from '../../helpers';
+
+describe('Create Booking Page', () => {
+  let bookingComponent, production, server, performanceModel, routerPushFake;
+  let fakeComponent = {
+    name: 'router-view',
+    template: '<div />',
+    props: ['production', 'booking', 'ticket_matrix'],
+  };
+  beforeAll(async () => {
+    server = await executeWithServer(async (server) => {
+      performanceModel = server.create(
+        'performanceNode',
+        FakePerformance(server)
+      );
+
+      let { data } = await runApolloQuery({
+        query: require('@/graphql/queries/ProductionBySlug.gql'),
+        variables: {
+          slug: performanceModel.production.slug,
+        },
+      });
+
+      production = data.production;
+    }, false);
+  });
+  afterAll(() => {
+    server.shutdown();
+  });
+  beforeEach(() => {
+    bookingComponent = shallowMount(Book, {
+      propsData: {
+        production: production,
+      },
+      mocks: {
+        $route: {
+          params: {},
+          meta: {
+            stage: stages[0],
+          },
+        },
+        $router: {
+          push: (routerPushFake = jest.fn()),
+        },
+      },
+      stubs: { RouterView: fakeComponent, 'router-link': true },
+    });
+  });
+
+  it('has a production banner', () => {
+    let banner = bookingComponent.findComponent(ProductionBanner);
+    expect(banner.exists()).to.be.true;
+    expect(banner.props('production')).to.eq(production);
+    expect(banner.props('showBuyTicketsButton')).to.eq(false);
+    expect(banner.props('showDetailedInfo')).to.eq(false);
+  });
+
+  it('has booking navigation', () => {
+    let bookingNavigation = bookingComponent.findComponent(BookingNavigation);
+    expect(bookingNavigation.exists()).to.be.true;
+    expect(bookingNavigation.props('currentStageIndex')).to.eq(0);
+    expect(bookingNavigation.props('maxAllowedStageIndex')).to.eq(0);
+    expect(bookingNavigation.props('production')).to.eq(production);
+    expect(bookingNavigation.props('booking')).to.eq(
+      bookingComponent.vm.booking
+    );
+  });
```

Comment:
is it woth testing when the sidebar navigation dissapears in mobile?

Reply:
This is more of a visual thing rather than controlled by code, probably best to leave to E2E testing